### PR TITLE
Added multiarch build support for Teleport

### LINKED ISSUE #3384: Request: ARM container builds
This is a request to provide ARM container builds on quay.io.  Quay now supports multi-arch container manifests (anouncement [here](https://www.redhat.com/en/blog/introducing-red-hat-quay-3-registry-your-linux-and-windows-containers)).

NOTE: This is related to my personal use of Teleport Community Edition, not my employer's use of Teleport Enterprise.

### PR DESCRIPTION
**Pull request by fheinecke**
This is part two of a series of PRs updating drone/dronegen to support multiarch builds.

The primary purpose of this PR is to add multiarch Teleport container images. This includes amd64, arm, and arm64 images under the `$MAJOR_VERSION` tag on each release.

Due to the complexity of how we handle building and publishing container images this PR includes a number of other changes:
* Removed old/unneeded Makefile targets that pertain to building images.
* Reduced the number of Dockerfiles that are involved with building images from four to one consistent, unified file.
* Moved all Teleport OSS/Enterprise/FIPS CI/CD related image building logic into dronegen and the new "unified" Dockerfile. Previously this logic was spread across two repos, two makefiles, a few dozen makefile targets, four Dockerfiles, Dronegen, and lots of manual entries in `.drone.yml`
* Updated makefiles to use the new Dockerfile
* Refactored several makefile targets to support local archive building, deb building, and image building from source

After this PR is merged I will open a final PR for Teleport Lab.

Related PR: 
* https://github.com/gravitational/teleport/pull/16688
* https://github.com/gravitational/teleport/pull/15514
* https://github.com/gravitational/teleport/issues/3384

Test runs prior to merging PR:
* Tag logic with empty container repo (v1.2.3-fred.11): https://drone.platform.teleport.sh/gravitational/teleport/17046
* Tag logic again with the same tag, via retag (v1.2.3-fred.11): https://drone.platform.teleport.sh/gravitational/teleport/17048
* Promote logic with exit due to prerelease (v1.2.3-fred.11): https://drone.platform.teleport.sh/gravitational/teleport/17050
* Promote logic with empty container repos (v1.2.3-fred.11): https://drone.platform.teleport.sh/gravitational/teleport/17049
* Promote logic with the same tag, via restart (v1.2.3-fred.11): https://drone.platform.teleport.sh/gravitational/teleport/17051
* Cron logic test (v1.2.4-fred.11): https://drone.platform.teleport.sh/gravitational/teleport/17059

### CODE REVIEW

**Comment by fheinecke** [comment]
Note to reviewers: I'll be deploying a series of commits to test this PR in Drone itself. For now please ignore any commits after https://github.com/gravitational/teleport/pull/17597/commits/87ef4cf4ba6c2b1ca8070329db5e8391deef2b20.

**Comment by fheinecke** [comment]
Found at least one logic bug related to building image at tag time. Working on it now.

**Review by r0mant — APPROVED** [review]
I don't have many comments as it's basically impossible to review the drone.yaml changes. I'm very concerned about its maintainability going forward.

Please make sure to test FIPS builds before merging. Especially the fact that they are indeed built against boringcrypto.

**Review by wadells — DISMISSED** [review]
Please do not merge until you've tested with role based access.

File: .drone.yml
Comment (logand22) [inline .drone.yml:7243]: The re-ordering creates a lot of noise.

File: dronegen/container_image_products.go
Comment (logand22) [inline dronegen/container_image_products.go:134]: If we are checking out the source do we need to download the dockerfile?

**Review by logand22 — APPROVED** [review]
From what I can tell things look okay. I usually just spot check the .drone.yml for things I expect to see. However, inspecting all of the changes is basically impossible and things have been abstracted so much in `dronegen`, it's hard to follow as well. 

You need to unset your testing values as there are a lot of values in here corresponding to the wrong ECR staging registry and other various incorrect values. 

Thanks for including tests but please make sure to test what both @r0mant and @wadells have mentioned before merging.

File: .drone.yml
Comment (wadells) [inline .drone.yml:8502]: We'll want to move to everything wuing roles with the real secrets before we merge.  I'm flagging this as one example of non-role based access with fake credentials, but I'm sure there are more.

File: .drone.yml
Comment (fheinecke) [reply .drone.yml:7243]: This isn't a reordering strictly speaking. It's a complete removal of the `promote-docker-ecr` and `promote-docker-quay` pipelines

File: .drone.yml
Comment (fheinecke) [reply .drone.yml]: This has been implemented, pending another full round of testing

File: dronegen/container_image_products.go
Comment (fheinecke) [reply dronegen/container_image_products.go:138]: Yes and no. When the pipeline runs in the context of a branch that is _not_ the one it's building then it needs to download the version-specific dockerfile. For example, when triggered by a cron job the pipeline runs in the context of the master branch, but the cron jobs update images built from other specific branches (`branch/v11`, `branch/v10`, etc.).

I can make this step only run when strictly needed however IMO it's a premature optimization that would add complexity and not gain us much. Do you think I should implement that change?

File: Makefile
Comment (fheinecke) [reply Makefile]: Reverted after finishing testing

**Comment by fheinecke** [comment]
@wadells @r0mant @logand22 This is ready for a (hopefully) final review. All testing changes have been reverted. See PR main comment for a link to drone test runs. Images are also available for testing here:
* https://quay.io/repository/fred_heinecke/teleport-operator
* https://quay.io/repository/fred_heinecke/teleport
* https://quay.io/repository/fred_heinecke/teleport-ent
* Public ECR in the teleport-dev AWS account
* Privte ECR in the teleport-dev AWS account in us-east-2

**Pull request merged**
